src/db/index.js

```diff
@@ -41,13 +44,41 @@ const dbFactory = opts => {
   }
 
   // queryS :: (String, [a]) -> Stream b
-  const queryS = (sql, vals=[]) =>
-    _((async () => {
-      const client = await pool.connect()
-      const stream = _(client.query(new QueryStream(sql, vals)))
-      stream.observe().done(() => client.release())
-      return stream
-    })()).flatten()
+  const queryS = (sql, vals=[]) => {
+    let client, cursor
+
+    const close = once(err1 => {
+      debug(`closing cursor${err1 ? ' on error' : ''}`)
+      cursor.close(err2 =>
+        client.release(err1 || err2)
+      )
+    })
+
+    const openCursor = klient => {
+      debug('opening cursor')
+      client = klient
+      cursor = client.query(new Cursor(sql, vals))
+
+      return _((push, next) => {
+        cursor.read(100, (err, rows) => {
+          if (err) {
+            push(err)
+            next()
```

Comment:
:thinking: If we get an error on read, do we really want to move next? Or should the stream end here? I haven't dealt with cursors in a long time, & never with postgres, but i fear we'd be in an unpredictable state here?

Reply:
Where this is used internally, we `stopOnError` and close the cursor.  But a few of the internal functions are exposed publicly, and we document that closing the cursor on error is something that must be done manually.  Are you thinking we should detect this error and close early, while still passing the error through to the reader?

And you do use cursors in Postgres whenever you reach for `pg-query-stream`.  It wraps `pg-cursor` under the hood, but does so in a way that leaks the cursor and connection if you stop pulling from the Node Stream.  🤦‍♂

Reply:
I feel the safer course of action is, if a cursor fails to read, that we emit an error & end every time. I'm not sure is safe to depend on the (internal) upstream to end for us.

Reply:
We can cleanup the cursor early for our errors, but the reader will still need to manually cleanup if they error.  Implementing the former will not avoid the latter.

Reply:
Is that true if we use `onFinished`?

https://github.com/articulate/hermes/commit/7b07467ddb6602d11df258f3b98ca3f6fcaffa73#commitcomment-38232375

Reply:
Oh, no... because the consumer never finishes. Sorry, it all just clicked.